### PR TITLE
[Term Entry] Java Strings: .replaceAll method

### PR DESCRIPTION
**Pull request by marymkohn**
### Java Strings: .replaceAll method
Created a term entry for Codecademy docs

### Type of Change
- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/strings/terms/replaceAll/replaceAll.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+This method replaces all the substrings in the string that match the given regular expression (`regex`) with the given replacement (`repl`).
+
+- It's important to note that certain characters in the replacement string, such as backslashes and dollar signs ($), may have special meanings, so it's recommended to use `Matcher.quoteReplacement(String)` to prevent these characters from affecting the results.
+
+
+Both the `.replace()` and `.replaceAll()` methods replace all occurrences of a matching string and accept `CharSequence` as input. However, the key difference is that `.replaceAll()` treats the given `CharSequence` as a *regular expression* and searches for `regex` matches, while `.replace()` treats the input as plain text.
+
+## Example
+
+The example below removes all whitespace
```
Comment (SSwiniarski) [inline content/java/concepts/strings/terms/replaceAll/replaceAll.md:36]:
```suggestion
The example below removes all whitespace:
```

File: content/java/concepts/strings/terms/replaceAll/replaceAll.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+`regex` represents the regular expression to which this string is to be matched.
+`repl` represents the string to be substituted for each match.
+
+
+This method replaces all the substrings in the string that match the given regular expression (`regex`) with the given replacement (`repl`).
+
+- It's important to note that certain characters in the replacement string, such as backslashes and dollar signs ($), may have special meanings, so it's recommended to use `Matcher.quoteReplacement(String)` to prevent these characters from affecting the results.
+
+
+Both the `.replace()` and `.replaceAll()` methods replace all occurrences of a matching string and accept `CharSequence` as input. However, the key difference is that `.replaceAll()` treats the given `CharSequence` as a *regular expression* and searches for `regex` matches, while `.replace()` treats the input as plain text.
```
Comment (SSwiniarski) [inline content/java/concepts/strings/terms/replaceAll/replaceAll.md:32]:
```suggestion
> **Note:** Both the [`.replace()`](https://www.codecademy.com/resources/docs/java/strings/replace) and `.replaceAll()` methods replace all occurrences of a matching string and accept `CharSequence` as input. However, the key difference is that `.replaceAll()` treats the given `CharSequence` as a *regular expression* and searches for `regex` matches, while `.replace()` treats the input as plain text.
```

**Review by SSwiniarski — COMMENTED** [review]
@marymkohn I did a first review and left a few comments.

**Review by SSwiniarski — APPROVED** [review]
@marymkohn Looks ready for a second review.

File: content/java/concepts/strings/terms/replaceAll/replaceAll.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+---
+Title: '.replaceAll()'
+Description: 'Replaces every matching subsequence in the input sequence with the specified replacement string.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Strings'
+  - 'Methods'
```
Comment (caupolicandiaz) [inline content/java/concepts/strings/terms/replaceAll/replaceAll.md:9]:
```suggestion
  - 'Methods'
  - 'Characters'
```

File: content/java/concepts/strings/terms/replaceAll/replaceAll.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The `String` class' `.replaceAll()` method replaces every matching subsequence in the input sequence with the specified replacement string.
```
Comment (caupolicandiaz) [inline content/java/concepts/strings/terms/replaceAll/replaceAll.md:15]:
```suggestion
The `String` class' **`.replaceAll()`** method replaces every matching subsequence in the input sequence with the specified replacement string.
```

File: content/java/concepts/strings/terms/replaceAll/replaceAll.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+## Syntax
+
+```pseudo
+string.replaceAll(String regex, String repl);
+```
+
+`regex` represents the [regular expression](https://www.codecademy.com/resources/docs/general/regular-expressions) to which this string is to be matched.
+`repl` represents the string to be substituted for each match.
+
+This method replaces all the substrings in the string that match the given regular expression (`regex`) with the given replacement (`repl`).
```
Comment (caupolicandiaz) [inline content/java/concepts/strings/terms/replaceAll/replaceAll.md:26]:
```suggestion

```

File: content/java/concepts/strings/terms/replaceAll/replaceAll.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+The `String` class' `.replaceAll()` method replaces every matching subsequence in the input sequence with the specified replacement string.
+
+## Syntax
+
+```pseudo
+string.replaceAll(String regex, String repl);
+```
+
+`regex` represents the [regular expression](https://www.codecademy.com/resources/docs/general/regular-expressions) to which this string is to be matched.
```
Comment (caupolicandiaz) [inline content/java/concepts/strings/terms/replaceAll/replaceAll.md:23]:
```suggestion
`regex` represents the [regular expression](https://www.codecademy.com/resources/docs/general/regular-expressions): a set of special characters and letters used for matching specific text.
```

File: content/java/concepts/strings/terms/replaceAll/replaceAll.md (hunk 5 above)
Comment (caupolicandiaz) [reply content/java/concepts/strings/terms/replaceAll/replaceAll.md:26]: this feels redundant

File: content/java/concepts/strings/terms/replaceAll/replaceAll.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+The example below removes all whitespace:
+
+```java
+String myString = "Dog\t  \r\rComputer\n\n  \r House \t Yes \n  ";
+
+String regex = "\\s";
+String replacement = "";
+
+String newString = myString.replaceAll(regex, replacement);
```
Comment (caupolicandiaz) [inline content/java/concepts/strings/terms/replaceAll/replaceAll.md:42]:
```suggestion
public class Main {
  public static void main(String[] args) {
    String myString = "Dog\t  \r\rComputer\n\n  \r House \t Yes \n  ";

    String regex = "\\s";   
    // The shorthand code \s represents spaces, tabs, and newline characters
    String replacement = "";

    String newString = myString.replaceAll(regex, replacement);
    System.out.println(newString);
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@marymkohn Overall, this looks good. I've added a few additional edits

**Review by caupolicandiaz — APPROVED** [review]
@marymkohn LGTM

**Comment by SSwiniarski** [comment]
Congrats @marymkohn on your new entry. You can see it live here: https://www.codecademy.com/resources/docs/java/strings/replaceAll

It looks like you're currently anonymous:
image

If you'd like to become public, make sure to add your GitHub username to your Codecademy profile and set your visibility to everyone.

cc: @caupolicandiaz

**Comment by marjoriekohn** [comment]
@SSwiniarski I recently underwent a legal name change and updated my GitHub and Codecademy usernames. Also, at the time of this contribution, my profile was private, but I’ve since made it public. Can we update this contribution to publicly reflect my current username on Codecademy Docs?

**Comment by SSwiniarski** [comment]
@marjoriekohn I don't know if we can do this. Any advice: @yangc95, @KTom101, @caupolicandiaz?

**Comment by KTom101** [comment]
@marjoriekohn Have you gone about following the steps of adding the new GitHub username to your Codecademy account? Since you only changed the username on the account not create a new account, there shouldn't be a problem with getting authorship. The steps are [here](uhttps://discuss.codecademy.com/t/my-contribution-is-live-but-im-not-being-shown-as-a-contributor-why-is-this/758036rl).

**Comment by marjoriekohn** [comment]
@KTom101 My information is up to date on Codecademy and my profile is public.

**Comment by yangc95** [comment]
@marjoriekohn Did you head over to [linked accounts](https://www.codecademy.com/account/linked_accts) in the[ link that was provided ](https://discuss.codecademy.com/t/my-contribution-is-live-but-im-not-being-shown-as-a-contributor-why-is-this/758036) just to double-check that

**Comment by yangc95** [comment]
@marjoriekohn double checking now, I see that the newer PR you pushed up is showing up properly. I noticed that the emails you used for this one versus the new entry are different. Can you double check what you have as your email? The easiest way is to make a change to this file, after which you would appear here as a contributor.

In the future, just make sure you're pushing up with the same email. You can double check the email with `git log` and use `git config` to view your email is accurate.

**Pull request merged**